src/databricks/labs/ucx/source_code/base.py

```diff
@@ -459,6 +459,19 @@ def read_text(path: Path, size: int = -1) -> str:
         return f.read(size)
 
 
+def safe_read_text(path: Path, size: int = -1) -> str:
```

Comment:
What do you think about making this `Option[str]` and returning `None` instead?

I think this is a little cleaner, particularly because it lets the caller distinguish between empty and missing, while treating them both as falsey if that's all it cares about. Testing also becomes easier.

Stepping back: exceptions and their handling in Python are beyond the scope of the type checker. However by expressing the abnormal condition with the return type, suddenly the type-checker _can_ verify that callers are handling the situation correctly.

To be clear, this is not a blocking question.